### ISSUE TITLE
Can't connect websocket to  'wss://relay.walletconnect.com'

### ISSUE THREAD

**Issue by Yeulmax** [WalletConnect/walletconnect-monorepo#1733]
**Describe the bug**

Hi, i'm a web developer currently searching a service for implementing multiple wallet to a Dapp for a client.
I came across your product which seems to fit.
However, i'm facing some big issues.

On some devices, i can't initiate a websocket connection to 'wss://relay.walletconnect.com'.

I get this error:
Error: WebSocket connection failed for URL: wss://relay.walletconnect.com?auth=eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJkaWQ6a2V5Ono2TWtzTFJYc0ZmWmhwdG1jVVZpdnA3c2hnVGtpSDZabzlTQjRRdUNpUUoxYlR6VyIsInN1YiI6IjU4ZWE3NThjMGEzZTU0ZThkMDViZmEyZGFmMDA0NjgxMTI0ZjM1MjgyYzA4NTY5MDFhZTE0ODM2NTUxNzk0NDciLCJhdWQiOiJ3c3M6Ly9yZWxheS53YWxsZXRjb25uZWN0LmNvbSIsImlhdCI6MTY3MDQ0NDYzMiwiZXhwIjoxNjcwNTMxMDMyfQ.Nl3Awe1-iPy0db5mi30b1FtPUqla7GLIwjqd9mvmkF77BkeVy7j2QoUFqSpqy6peaSRDGLVGBRd5zDjEm6kVCQ&projectId=f19f5c8e2b1ea7fbd382583761c167b3&ua=wc-2%2Fjs-2.1.3%2Flinux-firefox-107.0.0%2Fbrowser%3Areact-wallet.walletconnect.com

As you can see in the attachments, Firefox interprets it as a code 401:Unauthorized, however the same code (on the same network) on a different machine, works fine, so the 'projectId' seems good. 

This occurs on my own code (prototype) which only execute that:

    const signClient = await this.SignClient.init({
        projectId: projectId,
        metadata: {
            name: projectName,
            description: "Example Dapp",
            url: "#",
            icons: ["https://walletconnect.com/walletconnect-logo.png"],
        },
    });

I think the code is not faulty because it is also a problem on your demo app: https://react-wallet.walletconnect.com/

After some test on the demo app and personal prototype, on more than 12 devices and 20 browser the connection does not work in some devices and work on others, erratically.

We have not been able to identify a common point, same devices model on same browser produce different result...

**SDK Version**

 "@walletconnect/sign-client": "^2.1.4",

**To Reproduce**

1. Go to 'https://react-wallet.walletconnect.com/'

**Expected behavior**
- On prototype, initiation of 'SignClient'
- On demo app, launch of the demo app

**Screenshots**

![image](https://user-images.githubusercontent.com/28804288/206289679-b2a1b709-8791-4f67-b975-01a7cf611aab.png)

![image](https://user-images.githubusercontent.com/28804288/206290787-466fc9f0-f031-405b-965a-01d0f5d2f704.png)

![Screenshot_20221207-170529_Samsung_Internet](https://user-images.githubusercontent.com/28804288/206289686-08d29141-142e-40d5-b191-db73afad3164.jpg)

**Desktop (please complete the following information):**
All major OS and browser

**Smartphone (please complete the following information):**
Multiple devices, Android and IOS

**Comment by xzilja** [comment]
Transferred this issue to a more appropriate repository. Not able to replicate this issue at the moment on my devices. Are there any proxies / firewalls or something similar you might have set up that can be interfering with this? Perhaps a browser extension?

**Comment by antondalgren** [comment]
I have the same issue. It worked when I left the office yesterday but this morning I cant connect through my app. 

Running `react-app.walletconnect.com` on chrome in the same emulator as my app issues this request: 
> wss://relay.walletconnect.com/?auth=eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJkaWQ6a2V5Ono2TWtqaUozWTFEdk1uWjFkaVNBRTNOMmVWejlzWTJSckF2YVdqQmMyWVNKRjFQTCIsInN1YiI6ImRmMTI2MmFkNjRkZTE1OTJkN2Q0YWIyOTc1ZTU1ZDk0NTdkMGJhZjQ4N2Y5MWUwYWM3NjQ2ZmIyODlmZDdjMDMiLCJhdWQiOiJ3c3M6Ly9yZWxheS53YWxsZXRjb25uZWN0LmNvbSIsImlhdCI6MTY3MDM3NTY2MiwiZXhwIjoxNjcwNDYyMDYyfQ.-mYjiirlaV9_UVqJ-04TT1gqg5jyPve8vCYU5ISNWhFkBVNeFvYFXlIsdGU5MPZUw74gb-a7fUMDfgALJXiNDQ&projectId=e899c82be21d4acca2c8aec45e893598&ua=wc-2%2Fjs-2.1.3%2Fandroidos-chrome-101.0.0%2Fbrowser%3Areact-app.walletconnect.com

With the response of `failed: HTTP Authentication failed; no valid credentials available`

Going to the `react-wallet.walletconnect.com` on the emulator show the same error as described in the issue. 
> wss://relay.walletconnect.com/?auth=eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJkaWQ6a2V5Ono2TWttcUJmZG9DY3ZQTldNOGVOTkJER0RUcGN0VHB4cFpNaXhUcXRHYnFub1ZEeCIsInN1YiI6IjUyNWY5YTk1NjYyOTNmZjVmNWExM2YyZmM5MWRmZjY1MGEyNDM1ZTdkNzAwYjM4ZGVmNTk2MjIxNDIyMDM5MDIiLCJhdWQiOiJ3c3M6Ly9yZWxheS53YWxsZXRjb25uZWN0LmNvbSIsImlhdCI6MTY3MDM3NTk4MiwiZXhwIjoxNjcwNDYyMzgyfQ.ESTLI0s9RIOKUo7nOy42NFkhXuWB2Uyfx058HMuaUZ4JSdKd9zBWVS27QwgiiK3t1-t1CKvBfQ68V9vJNtIOBw&projectId=f19f5c8e2b1ea7fbd382583761c167b3&ua=wc-2%2Fjs-2.1.3%2Fandroidos-chrome-101.0.0%2Fbrowser%3Areact-wallet.walletconnect.com

With the response of `failed: HTTP Authentication failed; no valid credentials available`
<img width="424" alt="image" src="https://user-images.githubusercontent.com/9164583/206404886-9a7a0ac1-6d2e-43c3-800e-d51617043c84.png">

**Comment by xzilja** [comment]
Ty for reporting, gonna raise this with our team

**Comment by antondalgren** [comment]
Turns out the clock on my emulator had stopped :) Changing it to the current time solved the issue for me. 

Thanks,

**Comment by Yeulmax** [comment]
Thank's a lot @antondalgren ! The common point on faulty device was a difference of 1-2mn on the device's clock, solve the problem on my side too

**Comment by antondalgren** [comment]
Maybe we could get a better error message from the relay server to give us a better indication to what is wrong with the authentication process?

**Comment by antondalgren** [comment]
Encountering connectivity issues again... 

My physical device which i use on a daily basis cant connect to the relay system either through my app or the `react-wallet.walletconnect.com`. The time on my device is the same as the time on my computer, and I can go to `react-wallet.walletconnect.com` on the computer.

A colleague of mine also tried to visit the page, using iOS and google chrome, he connected successfully. Another colleague tried to visit the page on a android device using google chrome. Both he and me managed to end up with the error that has been described above. (Android users on google chrome)

Please advise @0xAsimetriq

**Comment by antondalgren** [comment]
The clock on my device must be in a 24h window less than current time to be able to connect. This seems to be the case across the board for all devices.

**Comment by philmetzger** [comment]
I see this has been closed, but I have been able to reproduce this issue still by setting my clock back a few hours.
I havent seen any solutions though?

**Comment by chenguzhen87** [comment]
ｈｏｗ　ｔｏ　ｓｏｌｖｅ　ｔｈｉｓ　ｐｒｏｂｌｅｍ？

**Comment by domhhv** [comment]
Hi guys, just in case someone is running into this issue, it turned out that we had [Content-Security-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy) set up on our website.

So I managed to solve the issue on my end by enabling CSP's [connect-src](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/connect-src) to `*` on the page where I struggled to open the WalletConnect modal.

**Comment by hyeongju0109** [comment]
Re-adjust the time in Ubuntu

**Comment by vishalpql** [comment]
getting same error
""WebSocket connection failed for host: wss://relay.walletconnect.org""
any solution?

**Comment by maesterzak** [comment]
I am getting the same error "Error: WebSocket connection failed for host: wss://relay.walletconnect.com".

**Comment by Quadrizzz** [comment]
Error: WebSocket connection failed for host: wss://relay.walletconnect.org, I got the same error, does anyone know how to fix this

**Comment by Quadrizzz** [comment]
> Error: WebSocket connection failed for host: wss://relay.walletconnect.org, I got the same error, does anyone know how to fix this

[https://[react-wallet.walletconnect.com](https://react-wallet.walletconnect.com/)/](url) the same error is appearing on your app

**Comment by aur0104** [comment]
I was facing the same issue on iOS. The problem was caused by adding an App ID in the Reown (WalletConnect) dashboard. When you add an App ID, WalletConnect enables strict security filtering, meaning your wallet can only connect if:

- The iOS Bundle ID matches exactly
- The Android Package Name matches exactly
- The app origin is fully trusted

If any of these do not match, WalletConnect blocks the WebSocket upgrade request, which results in the connection error.
I fixed the issue by removing the App ID from the dashboard.
